### PR TITLE
Introduce tractusx-connector helm chart

### PR DESCRIPTION
**Pull request by DominikPinsel**

### CODE REVIEW

**Comment by DominikPinsel** [comment]
> Still in progress of reviewing. Also hadn't the chance to look at the latest commits. Hope it's already helpful!

Always :) Thanks! Still a lot in progress here, I created the draft mainly to try out the new business test pipeline.

File: charts/txdc/values.yaml [hunk 1]
```diff
@@ -0,0 +1,345 @@
+# Default values for eclipse-dataspace-connector.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+fullnameOverride: ""
+nameOverride: ""
+
+imagePullSecrets: [ ]
+
+customLabels: { }
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply charts/txdc/values.yaml:10]: Yes, it accepts it.

File: charts/txdc/values.yaml [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+    # We usually recommend not to specify default resources and to leave this as a conscious
+    # choice for the user. This also increases chances charts run on environments with little
+    # resources, such as Minikube. If you do want to specify resources, uncomment the following
+    # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+    # limits:
+    #   cpu: 100m
+    #   memory: 128Mi
+    # requests:
+    #   cpu: 100m
+  #   memory: 128Mi
```
Comment (florianrusch-zf) [inline charts/txdc/values.yaml:181]:
```suggestion
    #   memory: 128Mi
```

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]
Now I have looked at all the changes 🤣
Looks quite promising.

File: charts/txdc/templates/configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+kind: ConfigMap
+metadata:
+  name: {{ include "txdc.fullname" . }}-controlplane
+  namespace: {{ .Release.Namespace | default "default" | quote }}
+  labels:
+    {{- include "txdc.controlplane.labels" . | nindent 4 }}
+data:
+  opentelemetry.properties: |-
+    otel.javaagent.enabled=false
+    otel.javaagent.debug=false
```
[the remark this answers is not in the document]
Comment (denisneuling) [reply charts/txdc/templates/configmap.yaml:12]: Of course it must - but it's still a draft

File: charts/txdc/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+            #####################
+            ## DATA ENCRYPTION ##
+            #####################
+
+            # see extension https://github.com/catenax-ng/product-edc/tree/develop/edc-extensions/data-encryption
+            - name: EDC_DATA_ENCRYPTION_KEYS_ALIAS
+              value: {{ .Values.vault.secretNames.transferProxyTokenEncryptionAesKey | quote }}
+            - name: EDC_DATA_ENCRYPTION_ALGORITHM
+              value: AES
+
```
[the remark this answers is not in the document]
Comment (denisneuling) [reply charts/txdc/templates/deployment.yaml]: done.

File: charts/txdc/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+              value: {{ .Values.vault.azure.tenant | quote }}
+            - name: EDC_VAULT_NAME
+              value: {{ .Values.vault.azure.name | quote }}
+            - name: EDC_VAULT_CLIENTSECRET
+              value: {{ .Values.vault.azure.secret | quote }}
+            - name: EDC_VAULT_CERTIFICATE
+              value: {{ .Values.vault.azure.certificate | quote }}
+
+            {{- end }}
+
```
[the remark this answers is not in the document]
Comment (denisneuling) [reply charts/txdc/templates/deployment.yaml]: done.

File: charts/txdc/values.yaml [hunk 6]
```diff
@@ -0,0 +1,493 @@
...
+    client: ""
+    tenant: ""
+    secret: ""
+    certificate: ""
+  secretNames:
+    transferProxyTokenSignerPrivateKey: transfer-proxy-token-signer-private-key
+    transferProxyTokenSignerPublicKey: transfer-proxy-token-signer-public-key
+    transferProxyTokenEncryptionAesKey: transfer-proxy-token-encryption-aes-key
+    dapsPrivateKey: daps-private-key
+    dapsPublicKey: daps-public-key
```
Comment (florianrusch-zf) [inline charts/txdc/values.yaml:474]: I know, that the EDC setting was named `EDC_OAUTH_PUBLIC_KEY_ALIAS` but the input is just a certificate. Would it make sense to rename this property here to `daps-public-cert` or something similar to reduce the confusions for the operation people?

File: charts/txdc/values.yaml [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+  httpProxyTokenReceiverUrl: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: { }
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
```
[the remark this answers is not in the document]
Comment (florianrusch-zf) [reply charts/txdc/values.yaml:493]: @denisneuling what's you opinion about this?

File: charts/txdc/templates/deployment-controlplane.yaml [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+
+            # see extension https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/main/extensions/control-plane/store/sql/asset-index-sql
+            - name: "EDC_DATASOURCE_ASSET_NAME"
+              value: "asset"
+            - name: "EDC_DATASOURCE_ASSET_USER"
+              value: {{ .Values.postgresql.username | required ".Values.postgresql.username is required" | quote }}
+            - name: "EDC_DATASOURCE_ASSET_PASSWORD"
+              value: {{ .Values.postgresql.password | required ".Values.postgresql.password is required" | quote }}
+            - name: "EDC_DATASOURCE_ASSET_URL"
+              value: {{ .Values.postgresql.jdbcUrl | required ".Values.postgresql.jdbcUrl is required" | quote }}
```
Comment (florianrusch-zf) [inline charts/txdc/templates/deployment-controlplane.yaml:179]: The following is WIP!!

As far as I know, the database needs to be at the end of the jdbcUrl. E.g `jdbc:postgresql://my-postgres-postgresql:5432/asset`

Is it possible to leaf it away?? Can you double check this?

Within the documentation I found this (https://jdbc.postgresql.org/documentation/use/#connecting-to-the-database):

> The default is to connect to a database with the same name as the user name used to connect to the server.

Therefor I would say it's necessary to add the database name to the url

File: charts/txdc/values.yaml (hunk 6 above)
Comment (ndr-brt) [reply charts/txdc/values.yaml:474]: @florianrusch-zf I noted the same on upstream, I think it would be better to rename there also to avoid ambiguity

**Comment by denisneuling** [comment]
> As already talked about in the chat, I noticed that many of the URLs are built depending on whether there is an Ingress or not. Don't know if that's so smart, or if we should include the possibility to override it. We at ZF for example have the use case that we don't create ingresses for our connectors.

@florianrusch-zf This now has been implemented for the IDS-API of the control-plane and for the "public" API of the data-plane. Is this fine enough or do we need to have this functionality as well for the other ones (data-management, control, validation etc)?

**Comment by florianrusch-zf** [comment]
> @florianrusch-zf This now has been implemented for the IDS-API of the control-plane and for the "public" API of the data-plane. Is this fine enough or do we need to have this functionality as well for the other ones (data-management, control, validation etc)?

@denisneuling fine for me ;-)

**Review by florianrusch-zf — APPROVED** [review]

File: charts/txdc-connector/Chart.yaml [hunk 9]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v2
-name: txdc
+name: txdc-connector
```
Comment (florianrusch-zf) [inline charts/txdc-connector/Chart.yaml:2]: Was thinking that `txdc` was standing for `Tractus-X Dataspace Connector`. So we would now have `Tractus-X Dataspace Connector connector` 😅 which makes no sense.

What is the meaning of the `txdc` abbreviation?

Comment (DominikPinsel) [reply charts/txdc-connector/Chart.yaml]: yes :) its now tractusx-connector

**Pull request merged**